Utilities/StaticAnalyzers/src/ArgSizeChecker.cpp

```diff
@@ -70,11 +70,11 @@ namespace clangcms {
 
           const clang::ento::PathDiagnosticLocation DLoc =
               clang::ento::PathDiagnosticLocation::createBegin(PVD, ctx.getSourceManager());
-
-          std::unique_ptr<BugType> BT = std::make_unique<clang::ento::BugType>(
-              this, "Function parameter copied by value with size > max", "ArgSize");
+          if (!BT1)
+            BT1 = std::make_unique<clang::ento::BugType>(
+                this, "Function parameter copied by value with size > max", "ArgSize");
           std::unique_ptr<BasicBugReport> report =
-              std::make_unique<BasicBugReport>(*BT, llvm::StringRef(os.str()), DLoc);
+              std::make_unique<BasicBugReport>(*BT1, llvm::StringRef(os.str()), DLoc);
```

Comment:
Assuming the `BasicBugReport` does not take address of the first argument, and that `clang::ento::BugType` is not a heavy type, how about something along
```cpp
clang::ento::BugType bt1(this, "Function parameter copied by value with size > max", "ArgSize");
auto report = std::make_unique<BasicBugReport>(std::move(bt1), llvm::StringRef(os.str()), DLoc);
```
or
```suggestion
              std::make_unique<BasicBugReport>(clang::ento::BugType(this, "Function parameter copied by value with size > max", "ArgSize"),
                                               llvm::StringRef(os.str()), DLoc);
```
?

Reply:
OK

Reply:
Memory usage hits the limits on cmsdev25 when I converted all of the BugType constructors to this.

Reply:
Yeah, after looking at it I see `clang::ento::BugReport` (base class of `BasicBugReport`) constructor taking  the address of the first argument
https://clang.llvm.org/doxygen/BugReporter_8h_source.html
so indeed the earlier you had is needed. Sorry for the noise and extra work.